framework/src/base/Adaptivity.C

```diff
@@ -133,7 +135,7 @@ bool
 Adaptivity::adaptMesh()
 {
   bool meshChanged = false;
-  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time))
+  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time) && _step % _interval == 0)
```

Comment:
This might be the right place for this, unfortunately though do to poor implementation, we need to now do something about the logic here: https://github.com/idaholab/moose/blob/devel/framework/src/base/FEProblem.C#L3681-L3696

With this change, MOOSE will print that adaptivity is happening but then not actually do it if the interval doesn't line up.